src/index.js

```diff
@@ -77,11 +82,7 @@ const mapToPropertyDescriptor = (
         return this.hasAttribute(name);
       },
       set(value) {
-        if (value) {
-          this.setAttribute(name, '');
-        } else {
-          this.removeAttribute(name);
-        }
+        setBooleanAttribute(this, name, value);
```

Comment:
it's a generic helper ... so I don't differentiate between pure and impure helpers/utils